Reworked codechef scraper

LaRuim opened the pull request:
Added a waiter to codechef scraper to replace while loop; converted the scraper into function, with a view to call it from an executor python file.

Added utility file for minor functions such as logging, rankfile naming, etc;

LaRuim committed:
Added utilities file to smoothen code; Reworked codechef scraper to include waiter, and name the rankfile

souravtecken reviewed (commented):
@LaRuim, What exactly is the workflow here? The entire utitlities file seems to be specific for the codechef scraper. Also, since this is a step towards unifying the scrapers, can't we handle the file writing in the common script, this method would involve having file handling operations in all scrapers, we could instead have the scraper return the data, right? 

Also, use pythonic naming conventions? xD, I'm a fan of camel case too, but the rest of the repo follows the pythonic conventions, it's better to stick with it.

LaRuim committed:
Removed camel case; split utilities into two files; codechef scraper now returns data instead of writing to file directly.

LaRuim committed:
Removed camel case; split utilities into two files; codechef scraper now returns data instead of writing to file directly.

LaRuim commented:
Right; utiltities was a file I created for convenience when working with selenium before, and I added functions I thought might be pertinent to the scraper there as well. In any case, I've split it based on functionality (into selenium_utils.py and log.py), and moved the codechef-specific functions to the scraper itself. As to why there is still a selenium_utils file; In my opinion, adding it to the scraper sorta bloats the code unnecessarily, but the functionality is still useful. 

The scraper now returns the data as a list, and the file handling will be done in the common script, which I will commit once you approve this

souravtecken commented on utils/selenium_utils.py:
```diff
@@ -0,0 +1,44 @@
+from selenium.webdriver.support import expected_conditions as EC
+from time import strftime, sleep
+
+month = ['jan', 'feb', 'march', 'april', 'may', 'june', 'july', 'aug', 'sept', 'oct', 'nov', 'dec']
```
Place this in the unified scrapers file

souravtecken commented on scrapers/codechef.py:
```diff
@@ -1,57 +1,117 @@
+        final_scoreboard.extend(divisions['B'].scraped_scoreboard) # Consolidate scoreboard
+        final_scoreboard = sorted(final_scoreboard,key=lambda x:x[1],reverse=True) # Sort list in desc. order based on points. (username, points)
+
+        if contest_code[0:5] == 'LTIME':
```
This should probably be moved to the unified scrapers file too, let the only job of the scraper file be to fetch the rankings, offload everything else to the main file. The task of writing has already been offloaded to that, so I think this should too.

LaRuim replied:
this also is a codechef-specific thing; none of the other platforms require intricate renaming..

LaRuim replied:
okay; will do anyway

souravtecken commented on scrapers/codechef.py:
```diff
@@ -1,57 +1,117 @@
+    print()
+    driver.close()
+
+def find_month_year(offSet, ID, start):
```
Some documentation required here, and move this to the main file too

LaRuim replied:
this is used only for the codechef scraper though

LaRuim committed:
Made changes requested; moved code to main file (yet to commit)

LaRuim commented:
@souravtecken excusez moi
